### DISCUSSION TITLE
Como rodar a partir do fonte? Preciso rodar o forms-flow a partir do código fonte?

### DISCUSSION THREAD

**Discussion by Emanuellv** [AOT-Technologies/forms-flow-ai#1874] [Installation]
Como rodar a partir do fonte? Preciso rodar o forms-flow a partir do código fonte, pois preciso fazer modificações no componente web. Mas não estou entendendo o mecanismo. Eu rodo o componente web na porta 3004 e o web-root-config na 3000. Quando abro no navegador a porta 3000 , ele nao renderiza o conteúdo e se eu tentar fazer alterações no componente web não renderiza.  Qual passo a passo para uma instalação a partir do fonte?
![image](https://github.com/AOT-Technologies/forms-flow-ai/assets/81375485/93a9c263-8a40-4ae0-ade4-9705128edc25)

**Comment by abilpraju-aot** [comment]
You can clone the main repository from here
[main-repository](https://github.com/AOT-Technologies/forms-flow-ai
)There after you can do the full deployment from docker, for that first you need to set up the keycloak 
Refer [Keycloak-deployment](https://github.com/AOT-Technologies/forms-flow-ai/blob/develop/forms-flow-idm/keycloak/README.md)
There after need to do the full deployment from docker refer 
[docker-full-deployment](https://github.com/AOT-Technologies/forms-flow-ai/tree/develop/deployment/docker 
)
And first change the sample env to .env 
And change necessary environment variables
And once all up then you need to customize the web component, for that you can stop the web root config running on docker 

move to web-component folder and  install the Npm dependencies using 
npm i and there after you can start by 
npm start
Once web is started you need to start the web-root-config for that install necessary npm dependencies by using npm i and then change the sample .env to .env and change the MF variable for the web component to 

MF_FORMSFLOW_WEB_URL=//localhost:3004/forms-flow-web.js

once that done And  change the config file located on forms-flow-web-root-config->public->config->config.js with below code
window["_env_"] = {
    "NODE_ENV": "development",
    "REACT_APP_API_SERVER_URL": "http://{your-ip-address}:3001",
    "REACT_APP_API_PROJECT_URL": "http://{your-ip-address}:3001",
    "REACT_APP_KEYCLOAK_CLIENT": "forms-flow-web",
    "REACT_APP_KEYCLOAK_URL_REALM": "forms-flow-ai",
    "REACT_APP_KEYCLOAK_URL": "http://{your-ip-address}:8080",
    "REACT_APP_WEB_BASE_URL": "http://{your-ip-address}:5000",
    "REACT_APP_BPM_URL": "http://{your-ip-address}:8000/camunda",
    "REACT_APP_WEBSOCKET_ENCRYPT_KEY": "giert989jkwrgb@DR55",
    "REACT_APP_APPLICATION_NAME": "formsflow.ai",
    "REACT_APP_WEB_BASE_CUSTOM_URL": "",
    "REACT_APP_CUSTOM_SUBMISSION_URL": "http://{your-ip-address}:{port}",
    "REACT_APP_CUSTOM_SUBMISSION_ENABLED": "false",
    "REACT_APP_ENABLE_APPLICATION_ACCESS_PERMISSION_CHECK": "false",
    "REACT_APP_MULTI_TENANCY_ENABLED": "false",
    "REACT_APP_DRAFT_ENABLED": "false",
    "REACT_APP_DRAFT_POLLING_RATE": "15000",
    "REACT_APP_EXPORT_PDF_ENABLED": "false",
    "REACT_APP_PUBLIC_WORKFLOW_ENABLED": "false",
    "REACT_APP_DOCUMENT_SERVICE_URL": "",
    "REACT_APP_CUSTOM_THEME_URL": "",
    "REACT_APP_CUSTOM_RESOURCE_BUNDLE_URL": "",
    "REACT_APP_MT_ADMIN_BASE_URL": "",
    "REACT_APP_KEYCLOAK_ENABLE_CLIENT_AUTH": "false",
    "REACT_APP_ENABLE_FORMS_MODULE": "true",
    "REACT_APP_ENABLE_TASKS_MODULE": "true",
    "REACT_APP_ENABLE_DASHBOARDS_MODULE": "true",
    "REACT_APP_ENABLE_PROCESSES_MODULE": "true",
    "REACT_APP_ENABLE_APPLICATIONS_MODULE": "true",
    "REACT_APP_DATE_FORMAT": "DD-MM-YY",
    "REACT_APP_TIME_FORMAT": "hh:mm:ss A"
}
Make sure all env variables are properly configured on the web-root-config folder
Once all done start the web-root-config by npm start 
there after you can see the changes that you done on web will reflect on the project
